Some extensions fail with unique key violation

### Steps to reproduce

1. Install DM5 (中文 language)
2. Enable comments (first toggle)
3. Open any title, e.g. 妖神记 or 花薰凛然
4. With high probablity, an error is presented

### Expected behavior

The pages to load

### Actual behavior

```
suwayomi-1  | 13:16:21.458 [DefaultDispatcher-worker-2] ERROR suwayomi.tachidesk.graphql.AsDataFetcherResult -- asDataFetcherResult: failed due to
suwayomi-1  | org.jetbrains.exposed.exceptions.ExposedSQLException: org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PUBLIC.UC_PAGE_INDEX_B ON PUBLIC.PAGE(INDEX, CHAPTER) VALUES 40186"; SQL statement:
suwayomi-1  | INSERT INTO PAGE (INDEX, URL, IMAGE_URL, CHAPTER) VALUES (?, ?, ?, ?) [23505-200]
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:99)
suwayomi-1  | 	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:292)
suwayomi-1  | 	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:269)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.Statement.execute(Statement.kt:59)
suwayomi-1  | 	at org.jetbrains.exposed.sql.QueriesKt.executeBatch(Queries.kt:447)
suwayomi-1  | 	at org.jetbrains.exposed.sql.QueriesKt.batchInsert(Queries.kt:357)
suwayomi-1  | 	at org.jetbrains.exposed.sql.QueriesKt.batchInsert(Queries.kt:332)
suwayomi-1  | 	at org.jetbrains.exposed.sql.QueriesKt.batchInsert$default(Queries.kt:327)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownload.updateDownloadStatusAndPageList$lambda$0$0(ChapterForDownload.kt:167)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:355)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$lambda$11(ThreadLocalTransactionManager.kt:402)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:401)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$lambda$5(ThreadLocalTransactionManager.kt:310)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:259)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:237)
suwayomi-1  | 	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:236)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownload.updateDownloadStatusAndPageList(ChapterForDownload.kt:159)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownload.asDownloadReady(ChapterForDownload.kt:86)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownloadKt.getChapterDownloadReady(ChapterForDownload.kt:40)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownloadKt.getChapterDownloadReady$default(ChapterForDownload.kt:34)
suwayomi-1  | 	at suwayomi.tachidesk.manga.impl.chapter.ChapterForDownloadKt.getChapterDownloadReadyById(ChapterForDownload.kt:43)
suwayomi-1  | 	at suwayomi.tachidesk.graphql.mutations.ChapterMutation$fetchChapterPages$1.invokeSuspend(ChapterMutation.kt:298)
suwayomi-1  | 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
suwayomi-1  | 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
suwayomi-1  | 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
suwayomi-1  | 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
suwayomi-1  | 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
suwayomi-1  | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
suwayomi-1  | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
suwayomi-1  | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
suwayomi-1  | Caused by: org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PUBLIC.UC_PAGE_INDEX_B ON PUBLIC.PAGE(INDEX, CHAPTER) VALUES 40186"; SQL statement:
suwayomi-1  | INSERT INTO PAGE (INDEX, URL, IMAGE_URL, CHAPTER) VALUES (?, ?, ?, ?) [23505-200]
suwayomi-1  | 	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1235)
suwayomi-1  | 	at com.zaxxer.hikari.pool.ProxyStatement.executeBatch(ProxyStatement.java:128)
suwayomi-1  | 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeBatch(HikariProxyPreparedStatement.java)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.jdbc.JdbcPreparedStatementImpl.executeBatch(JdbcPreparedStatementImpl.kt:97)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.InsertStatement.execInsertFunction(InsertStatement.kt:232)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:244)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:19)
suwayomi-1  | 	at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:97)
suwayomi-1  | 	... 31 common frames omitted
```

### Suwayomi-Server version

v2.1.1941

### Server operating system

docker

### Server Desktop Environment

_No response_

### Server JVM version

as in docker

### Used client name

Suwayomi-WebUI

### Client version

r2823

### Used web browser

Firefox

### Client operating system

Ubuntu 24.04

### Other details

Enabling comments generates a lot of pages, sometimes hundreds (compared to ~20 pages of actual content). I've seen that chapters with 200 pages load.

When loading the same manga in Mihon, everything loads fine. In particular, two of those tested have only few comment pages (32 total with 22 content).

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [x] I have written a short but informative title (ideally less than ~100 characters).
- [x] I have tried the troubleshooting guide described in [README.md](https://github.com/Suwayomi/Suwayomi-Server?tab=readme-ov-file#troubleshooting-and-support)
- [x] I have updated to the **[latest version](https://github.com/suwayomi/suwayomi-server/releases/latest)**.
- [x] I have filled out all of the requested information in this form, including specific version numbers.
- [x] I understand that **Suwayomi does not have or fix any extensions**, and I **will not receive help** for any issues related to sources or extensions.

--- a/server/src/main/kotlin/suwayomi/tachidesk/manga/impl/chapter/ChapterForDownload.kt
+++ b/server/src/main/kotlin/suwayomi/tachidesk/manga/impl/chapter/ChapterForDownload.kt
@@ -194,14 +194,15 @@ private class ChapterForDownload(
         val mangaEntry = transaction { MangaTable.selectAll().where { MangaTable.id eq mangaId }.first() }
         val source = getCatalogueSourceOrStub(mangaEntry[MangaTable.sourceReference])
 
-        return source.getPageList(
-            SChapter.create().apply {
-                url = chapterEntry[ChapterTable.url]
-                name = chapterEntry[ChapterTable.name]
-                scanlator = chapterEntry[ChapterTable.scanlator]
-                chapter_number = chapterEntry[ChapterTable.chapter_number]
-                date_upload = chapterEntry[ChapterTable.date_upload]
-            },
-        )
+        return source
+            .getPageList(
+                SChapter.create().apply {
+                    url = chapterEntry[ChapterTable.url]
+                    name = chapterEntry[ChapterTable.name]
+                    scanlator = chapterEntry[ChapterTable.scanlator]
+                    chapter_number = chapterEntry[ChapterTable.chapter_number]
+                    date_upload = chapterEntry[ChapterTable.date_upload]
+                },
+            ).mapIndexed { index, page -> Page(index, page.url, page.imageUrl, page.uri) }
     }
 }

